orbit/pkg/table/windowsupdatetable/windowsupdate.go

```diff
@@ -55,16 +62,23 @@ func TablePlugin(mode tableMode, logger zerolog.Logger) *table.Plugin {
 	return table.NewPlugin(t.name, columns, t.generate)
 }
 
-func queryUpdates(searcher *windowsupdate.IUpdateSearcher) (interface{}, error) {
-	return searcher.Search("Type='Software'")
+func queryUpdates(searcher WindowsUpdatesSearcher) (interface{}, error) {
+	searchResult, err := searcher.Search("Type='Software' AND IsInstalled=0")
+	if err != nil {
+		return nil, err
+	}
+
+	// We only care about the results iff we got some updates
+	if searchResult != nil && len(searchResult.Updates) == 0 {
+		return nil, nil
+	}
```

Comment:
Why is this needed? (What happens when `queryUpdates` returns `nil`?)

Reply:
When querying the `windows_updates` table we want to return no search results if there are no pending updates which is achieved here by returning nil if no updates match the specified criteria.

Bear in mind that that `searchResult` is a struct that contains the queried windows updates (inside the `Updates` prop) the absence of updates matching the criteria doesn't imply that searchResult will be nil. 

IMHO it would make more sense if the thing that gets serialized back is `searchResult.Updates` and we make it so that there's a 1:1 relationship between each `searchResult.Updates` element and each search result row, but that's outisde the scope of this and it is a breaking change (because the 'structure' the results would change).